HACKING.rst

```diff
@@ -1,263 +1,512 @@
+   brew install pyenv-virtualenv
 
-Installing Salt for development
--------------------------------
+--------------
 
-Clone the repository using::
+Now add pyenv to your ``.bashrc``:
 
-    git clone https://github.com/saltstack/salt
-    cd salt
+::
 
-.. note:: tags
+   echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
+   pyenv init 2>> ~/.bashrc
+   pyenv virtualenv-init 2>> ~/.bashrc
 
-    Just cloning the repository is enough to work with Salt and make
-    contributions. However, you must fetch additional tags into your clone to
-    have Salt report the correct version for itself. To do this, fetch the tags
-    with the command::
+For other shells, see `the pyenv
+instructions <https://github.com/pyenv/pyenv#basic-github-checkout>`__.
 
-        git fetch --tags
+Go ahead and restart your shell. Now you should be able to install a new
+version of Python:
 
-Preparing your system
-~~~~~~~~~~~~~~~~~~~~~
+::
+
+   pyenv install 3.7.0
+
+If that fails, don’t panic! You’re probably just missing some build
+dependencies. Check out `pyenv common build
+problems <https://github.com/pyenv/pyenv/wiki/Common-build-problems>`__.
+
+Now that you’ve got your version of Python installed, you can create a
+new virtual environment with this command:
+
+::
+
+   pyenv virtualenv 3.7.0 salt
+
+Then activate it:
+
+::
+
+   pyenv activate salt
+
+Sweet! Now you’re ready to clone salt so you can start hacking away!
+
+Cloning Salt
+~~~~~~~~~~~~
+
+Clones are so shallow. Well, this one is anyway:
+
+::
+
+   git clone --depth=1 --origin salt https://github.com/saltstack/salt.git
+
+This creates a shallow clone of salt, which should be fast. Most of the
+time that’s all you’ll need, and you can start building out other
+commits as you go. If you *really* want all 108,300+ commits you can
+just run ``git fetch --unshallow``. Then go make a sandwich because it’s
+gonna be a while.
+
+Fork Salt
+~~~~~~~~~
+
+You’re also going to want to head over to GitHub and create your own
+`fork of salt <https://github.com/saltstack/salt/fork>`__. Once you’ve
+got that setup you can add it as a remote:
+
+::
+
+   git remote add yourname <YOUR SALT REMOTE>
+
+If you use your name to refer to your fork, and ``salt`` to refer to the
+official Salt repo you’ll never get ``upstream`` or ``origin`` confused.
+
+Install sphinx
+~~~~~~~~~~~~~~
+
+In order to build the docs, you’ll need to install sphinx into your
+virtual environment:
+
+::
+
+   python -m pip install sphinx
+
+We’ll use this later.
+
+``pre-commit`` Setup
+~~~~~~~~~~~~~~~~~~~~
+
+Here at Salt we use `pre-commit <https://pre-commit.com/>`__ to make it
+easier for contributors to get quick feedback. It configures some Git
+pre-commit hooks to run ``black`` for formatting, ``isort`` for keeping
+our imports sorted, and ``pylint`` to catch issues like unused imports,
+among others. You can easily install it in your virtualenv with:
+
+::
+
+   python -m pip install pre-commit
+   pre-commit init
```

Comment:
see thread https://saltstackcommunity.slack.com/archives/C8832QN4U/p1605628370480700?thread_ts=1605085948.461300&cid=C8832QN4U. should be `pre-commit install`